src/library/scala/collection/Iterator.scala

```diff
@@ -635,14 +635,7 @@ trait Iterator[+A] extends IterableOnce[A] with IterableOnceOps[A, Iterator, Ite
     def next() = if (hasNext) { hdDefined = false; hd } else Iterator.empty.next()
   }
 
-  def drop(n: Int): Iterator[A] = {
-    var i = 0
-    while (i < n && hasNext) {
-      next()
-      i += 1
-    }
-    this
-  }
+  def drop(n: Int): Iterator[A] = sliceIterator(n, -1)
```

Comment:
I have mixed feelings about this. on the one hand, it's a simplification. on the other hand, there has been some desire (can't recall where it was expressed originally) to change `drop` to return `this.type` once we can break forward bincompat. it's not clear to me that you could make the same guarantee of returning `this.type` for `sliceIterator`, and if not, we would no longer be able to make the improvement to `drop`'s signature in the future.

of course, there's also an argument that perhaps `drop` returning `this.type` precludes laziness, and laziness is a more important trait

Reply:
I don't remember discussion 7 yrs about that; the previous fix was because the previous accommodation of laziness was not stack-safe. Also, I didn't see any discussion around reverting `take` to lazy. My only intuition is that `drop` and `take` should behave the same. Thanks for taking a look!